mesonbuild/environment.py

```diff
@@ -312,13 +312,26 @@ def __init__(self, source_dir, build_dir, options):
             # Used by the regenchecker script, which runs meson
             self.coredata.meson_command = mesonlib.meson_command
             self.first_invocation = True
-        self.cross_info = None
         self.exe_wrapper = None
+
+        self.machines = MachineInfos()
+        # Will be fully initialized later using compilers later.
+        self.machines.detect_build()
         if self.coredata.cross_file:
             self.cross_info = CrossBuildInfo(self.coredata.cross_file)
             if 'exe_wrapper' in self.cross_info.config['binaries']:
                 from .dependencies import ExternalProgram
                 self.exe_wrapper = ExternalProgram.from_cross_info(self.cross_info, 'exe_wrapper')
+            if 'host_machine' in self.cross_info.config:
+                self.machines.host = MachineInfo.from_literal(
+                        self.cross_info.config['host_machine'])
```

Comment:
__[Flake8]__

[E126] continuation line over-indented for hanging indent


<!-- Comment by [Sider](https://sider.review) -->